sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -206,6 +212,48 @@ def test_partial_dependence_helpers(est, method, target_feature):
     assert np.allclose(pdp, mean_predictions, rtol=rtol)
 
 
+@skip_if_32bit
+def test_decision_tree_vs_gradient_boosting():
+    # Make sure that the recursion method gives the same results on a
+    # DecisionTreeRegressor and a GradientBoostingRegressor with 1 tree and
+    # same parameters. The DecisionTreeRegressor doesn't pass the
+    # test_partial_dependence_helpers() test.
+
+    # Purely random dataset to avoid correlated features
+    n_samples = 100
+    n_features = 5
+    X = np.random.RandomState(0).randn(n_samples, n_features)
+    y = np.random.RandomState(0).randn(n_samples)
+
+    # The 'init' estimator for GBDT (here the average prediction) isn't taken
+    # into account with the recursion method, for technical reasons. We set
+    # the mean to 0 to that this 'bug' doesn't have any effect.
+    y = y - y.mean()
+
+    # set max_depth not too high to avoid splits with same gain but different
+    # features
+    max_depth = 5
+    gbdt = GradientBoostingRegressor(n_estimators=1, learning_rate=1,
+                                     criterion='mse', max_depth=max_depth,
+                                     random_state=0)
+    gbdt.fit(X, y)
+
+    tree = DecisionTreeRegressor(random_state=0, max_depth=max_depth)
+    tree.fit(X, y)
+
+    assert np.allclose(gbdt.predict(X), tree.predict(X))  # sanity check
+
+    grid = np.random.RandomState(0).randn(50).reshape(-1, 1)
+    for f in range(n_features):
+        features = np.array([f], dtype=np.int32)
+
+        pdp_gbdt = _partial_dependence_brute(gbdt, grid, features, X,
+                                             response_method='auto')
+        pdp_tree = _partial_dependence_brute(tree, grid, features, X,
```

Comment:
Why do you test with the brute force method? Why not the recursion method or the public `partial_dependence` function with auto mode?

Reply:
It should definitely be the recursion method, my bad